exporter/exporterhelper/internal/obs_report_sender.go

```diff
@@ -137,3 +162,38 @@ func toNumItems(numExportedItems int, err error) (int64, int64) {
 	}
 	return int64(numExportedItems), 0
 }
+
+func extractFailureAttributes(err error) attribute.Set {
+	if err == nil {
+		return attribute.NewSet()
+	}
+
+	attrs := []attribute.KeyValue{}
+	reason := determineFailureReason(err)
+	attrs = append(attrs, attribute.String(string(FailureReasonKey), reason))
+	isPermanent := consumererror.IsPermanent(err)
+	attrs = append(attrs, attribute.Bool(string(FailurePermanentKey), isPermanent))
+	return attribute.NewSet(attrs...)
+}
+
+func determineFailureReason(err error) string {
+	if err == nil {
+		return ""
+	}
+	if strings.Contains(err.Error(), "no more retries left") {
+		return "retries_exhausted"
+	}
+	if experr.IsShutdownErr(err) {
+		return "shutdown"
+	}
+	if consumererror.IsPermanent(err) {
+		return "permanent_error"
+	}
+	if errors.Is(err, context.Canceled) {
+		return "context_cancelled"
+	}
+	if errors.Is(err, context.DeadlineExceeded) {
+		return "context_timeout"
+	}
+	return "transient_error"
```

Comment:
```suggestion
	return "unknown"
```

If we remove "permanent_error" as suggested above and rely only on `failure.permanent`, then I don't think we can or should assume the error is transient. Rather, it's just that the reason is unknown; whether it's permanent/transient is orthogonal.